src/internet/radiobrowser/radiobrowserservice.h

```diff
@@ -0,0 +1,138 @@
+/* This file is part of Clementine.
+   Copyright 2010-2013, David Sansome <me@davidsansome.com>
+   Copyright 2010, 2014, John Maguire <john.maguire@gmail.com>
+   Copyright 2014, Krzysztof Sobiecki <sobkas@gmail.com>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef INTERNET_RADIOBROWSER_RADIOBROWSERSERVICE_H_
+#define INTERNET_RADIOBROWSER_RADIOBROWSERSERVICE_H_
+
+#include <QJsonObject>
+
+#include "core/cachedlist.h"
+#include "internet/core/internetmodel.h"
+#include "internet/core/internetservice.h"
+
+class RadioBrowserUrlHandler;
+
+class QNetworkAccessManager;
+class QNetworkReply;
+class QMenu;
+
+class RadioBrowserServiceBase : public InternetService {
```

Comment:
The base class probably isn't necessary unless you think other services will be derived from it. That said, there may be a place for a core StreamingService class when you look at code duplication.